### ISSUE TITLE
Use neural network to estimate state-value function

### ISSUE THREAD

**Issue by peter1591** [peter1591/hearthstone-ai#71]
1. input normalization
2. share weights for all minions
  * use convolution layer
  * or, partially connected layer
3. follow by 2-3 fully-connected layer

Library candidates:
Caffe --> many dependency? cross-platform to windows?
tiny-dnn --> header-only; no dependency; maybe slow
tensorflow --> mainly python support

**Comment by peter1591** [comment]
Use Adam optimizer

**Comment by peter1591** [comment]
finished in a973eda143d80d52140e21e48f2104aae1cfd61c
